joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/RubySrc2Cpg.scala

```diff
@@ -28,23 +28,34 @@ class RubySrc2Cpg extends X2CpgFrontend[Config] {
     withNewEmptyCpg(config.outputPath, config: Config) { (cpg, config) =>
       new MetaDataPass(cpg, Languages.RUBYSRC, config.inputPath).createAndApply()
       new ConfigFileCreationPass(cpg).createAndApply()
-      Using.resource(new ResourceManagedParser(config.antlrCacheMemLimit)) { parser =>
-        if (config.enableDependencyDownload && !scala.util.Properties.isWin) {
-          val tempDir = File.newTemporaryDirectory()
-          try {
-            downloadDependency(config.inputPath, tempDir.toString())
-            new AstPackagePass(cpg, tempDir.toString(), global, parser, RubySrc2Cpg.packageTableInfo, config.inputPath)(
-              config.schemaValidation
-            ).createAndApply()
-          } finally {
-            tempDir.delete()
-          }
-        }
+      if (config.useDeprecatedFrontend) {
+        deprecatedCreateCpgAction(cpg, config)
+      } else {
+        newCreateCpgAction(cpg, config)
+      }
+    }
+  }
 
-        val astCreationPass = new AstCreationPass(cpg, global, parser, RubySrc2Cpg.packageTableInfo, config)
-        astCreationPass.createAndApply()
-        TypeNodePass.withRegisteredTypes(astCreationPass.allUsedTypes(), cpg).createAndApply()
+  private def newCreateCpgAction(cpg: Cpg, config: Config): Unit = {
+    // TODO
```

Comment:
It looks like we're not yet triggering the new AST generator, or am I misreading this?

Reply:
You're correct. I thought of keeping the PR digestible, so I split this import into 2 steps: this first would only bring in the flag, and the next one would bring the new AstCreator